src/server/zset_family.cc

```diff
@@ -2443,12 +2448,61 @@ void ZSetFamily::ZRangeInternal(CmdArgList args, RangeParams range_params,
     }
   }
 
-  auto cb = [&](Transaction* t, EngineShard* shard) {
-    return OpRange(range_spec, t->GetOpArgs(shard), key);
+  OpResult<ScoredArray> range_result;
+  ShardId src_shard = Shard(key, shard_set->size());
+  auto range_cb = [&](Transaction* t, EngineShard* shard) {
+    if (shard->shard_id() != src_shard) {
```

Comment:
please check what should happen  if 
a) the destination key exists 
b) and is not a sorted set

Reply:
The destination is overriden (which seems to match the behaviour of Redis):
```
$ redis-cli
127.0.0.1:6379> SET foo bar
OK
127.0.0.1:6379> ZADD srczset 1 "one" 2 "two" 3 "three" 4 "four"
(integer) 4
127.0.0.1:6379> ZRANGESTORE foo srczset 2 -1
(integer) 2
127.0.0.1:6379> ZRANGE foo 0 -1
1) "three"
2) "four"
127.0.0.1:6379> 
```